Userland/Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -586,7 +586,7 @@ DynamicLoader::RelocationResult DynamicLoader::do_relocation(const ELF::DynamicO
         }
         VERIFY(dynamic_object_of_symbol);
         size_t addend = relocation.addend_used() ? relocation.addend() : *patch_ptr;
-        *patch_ptr = negative_offset_from_tls_block_end(dynamic_object_of_symbol->tls_offset().value(), symbol_value + addend);
+        *patch_ptr = addend + dynamic_object_of_symbol->tls_offset().value() + symbol_value;
```

Comment:
Any reason for dropping the `VERIFY(assert < 0)` ? Feels like it's still nice to have here.

Reply:
I originally planned to remove the whole "TLS offset is negative" part, but re-scoped my PR later (partially because I don't really understand the relocation code). Looks like the assert was the victim, which was simply forgotten about after I ripped out `negative_offset_from_tls_block_end` early. :^)

Reply:
I added the `VERIFY` back to the relocation side (although I don't fully understand how it works). When copying the TLS data during allocation, the condition should already be enforced by bounds-checks on the buffer (or rather, the fact that we are using a designated buffer at all).